spec/requests/hyku_addons/account_settings_request_spec.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe "::HykuAddons::AccountSettingsController", type: :request do
+  let(:user) { FactoryBot.create(:admin) }
+  let!(:account) { create(:account) }
+  before do
+    login_as(user, scope: :user)
+    allow(Apartment::Tenant).to receive(:switch).with(account.tenant) do |&block|
+      block.call
+    end
+    account.switch!
+  end
+
+  after do
+    account.reset!
+  end
+
+  describe 'GET /admin/account_settings' do
+    it "GETS index  /admin/account_settings" do
+      get admin_account_settings_url
+      expect(response).to have_http_status(:ok)
+    end
+  end
+
+  describe "Update single keys in settings hash" do
+    it "can update a single settings key eg  contact_email" do
+      patch update_single_admin_account_setting_url(id: account, partial_name: 'contact_email'), params: { "account" => { "settings" => { "contact_email" => "ab@abc.com" } } }
+      expect(response).to redirect_to admin_account_settings_url
+      follow_redirect!
```

Comment:
I am trying to mimic the exact thing that happens in the controller action, there is always a redirect after an update back to the index page which can contain many things, if you want another expect to check if a link or value exist on the page redirected to, i can add that.

Reply:
I was thinking the test could be something like this:
```
it "updates a single setting" do
  expect do
          patch update_single_admin_account_setting_path(id: account), params: { "account" => { "settings" => { "contact_email" => "ab@abc.com" } } }
  end.to change { Account.find(accout.id).settings.contact_email }.to('ab@abc.com")
  expect(response).to redirect_to admin_account_settings_path
end
```
This is what I'm doing a lot of right now with the API tests: test the request and what it's affects are then what the response looks like.

Reply:
If that is your preference, we can go with that.

Reply:
I will go with this:
```
  it "can update a single settings key eg  contact_email" do
    patch update_single_admin_account_setting_url(id: account), params: { "account" => { "settings" => { "contact_email" => "bo@abc.com" } } }
    account.reload
     expect(account.settings['contact_email']).to eq 'bo@abc.com'
      expect(response).to redirect_to admin_account_settings_url
  end
```

So I am  using account.reload instead of  Account.find(accout.id).